### PR TITLE
fix(codegen): fixes for esbuild test cases

### PR DESCRIPTION
**Pull request by Boshen**

### CODE REVIEW

**Comment by Boshen** [comment]
@Dunqing merge this after a oxc release first?

**Comment by Dunqing** [comment]
> @Dunqing merge this after a oxc release first?

We can merge this PR first. I saw a few performance improvements PRs don't merge. I want to wait for them

**Review by Dunqing — APPROVED** [review]

**Comment by Boshen** [comment]
> I want to wait for them

Let's ship those first and hold onto this one.

**Pull request merged**
